docs/src/contribute/design/mqtt-topic-design.md

```diff
@@ -0,0 +1,177 @@
+:::
+
+This section highlights the flexible nature of the MQTT topic schema which allows you to provide your own semantic meaning to each of the **identifier** segments within the topic schema.
+
+Let's start off by taking a closer look at the typical topic structure. The typical use-case defines the **identifier** (including root prefix) as follows:
+
+```mermaid
+graph LR
+  te --/--- segment1["device"] --/--- segment2["&lt;device_id&gt;"] --/--- segment3["service"] --/--- segment4["&lt;service_id&gt;"]
+```
+
+But what was not mentioned is that the two static segments, "device" and "service" are actually only used to provide the auto registration functionality, as the cloud mappers need to know how to map the topic to the specific digital twin entity (e.g. as a device, child device or service). The typical topic schema allows the mapper to infer this relationship purely form the topic (which is ok for the majority of use-cases). However, when you take away the auto registration functionality (e.g. disable it), it allows you to apply your own semantics to each of the 4-segment **identifier** providing more powerful modelling mechanisms.
+
+The **identifier** topic schema can be rewritten in its most generic form.
+
+```mermaid
+graph LR
+  te --/--- segment1["&lt;segment1&gt;"] --/--- segment2["&lt;segment2&gt;"] --/--- segment3["&lt;segment3&gt;"] --/--- segment4["&lt;segment4&gt;"]
+```
+
+With the generic form, the structure of the topics is still the same as before, 4 segments, however you can apply your own meaning to each of the segment by registering the entities (devices/services) at each of the specific segments. The mapper then matches the **identifier** used when publishing telemetry data or receiving commands, to the registered entity.
+
+This allows you to also encode specific information about the entity in the MQTT topic enabling powerful (local) filtering possibilities for other MQTT clients also running on the edge.
+
+Below is a simple example showing how the **identifier** can be modelled to represent the location of equipment in a factory. The custom **identifier** schema is utilized as follows:
+
+```mermaid
+graph LR
+  te --/--- segment1["&lt;factory&gt;"] --/--- segment2["&lt;building&gt;"] --/--- segment3["&lt;area&gt;"] --/--- segment4["&lt;equipment&gt;"]
+```
+
+The equipment, which is a conveyor belt called "belt01", is located in factory "factory01", in the "hallA" building, in the "packaging" area. The conveyor belt can be registered as a child-device by publishing the following message.
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/factory01/hallA/packaging/belt001' '{
+  "@type": "child-device",
```

Comment:
Aren't we missing the `"@parent"`  of this child device ?

Reply:
In the example `"@parent"` is not necessarily required, as it just means the `belt001` is an immediate child of the main device.